### PR TITLE
Rename `CheckCategory` to `RuleOrigin`

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Comment by charliermarsh** [comment]
\cc @not-my-profile

**Comment by not-my-profile** [comment]
Maybe rather `RuleOrigin` to clarify what origin is meant.

File: src/registry.rs [hunk 1]
```diff
@@ -888,399 +889,399 @@ impl DiagnosticCode {
         }
     }
 
-    pub fn category(&self) -> CheckCategory {
+    pub fn category(&self) -> DiagnosticOrigin {
```
Comment (not-my-profile) [inline src/registry.rs:892]: This function should be renamed as well to `origin`.

**Comment by not-my-profile** [comment]
Not related to this PR but why do the URLs in `CheckCategory::url` contain the version of the package? I don't think we actually keep track of which version our rules are based on, do we?

**Comment by charliermarsh** [comment]
That's the version at the time of porting, so it represents the version that was implemented in Ruff. (If those plugins evolve in future versions, we may deviate depending on how they change.)

**Pull request merged**